CredentialRequestOptions gains a "digital" member

marcoscaceres opened the pull request:
#### 3850dd46c4e4d996c6728ba993dbcbd639281d26
<pre>
CredentialRequestOptions gains a &quot;digital&quot; member
<a href="https://bugs.webkit.org/show_bug.cgi?id=270196">https://bugs.webkit.org/show_bug.cgi?id=270196</a>
<a href="https://rdar.apple.com/123744112">rdar://123744112</a>

Reviewed by Anne van Kesteren.

Adds &quot;digital&quot; member to CredentialRequestOptions dictionary:
<a href="https://wicg.github.io/digital-identities/#extensions-to-credentialrequestoptions-dictionary">https://wicg.github.io/digital-identities/#extensions-to-credentialrequestoptions-dictionary</a>

Given the spec changed names again, changed pref name to DigitalIdentityEnabled and updated a few files to reflect the name change.

* LayoutTests/http/wpt/credential-management/setDigitalCredentialsEnable.https-expected.txt: Copied from LayoutTests/http/wpt/credential-management/setDigitalIdentityEnable.https-expected.txt.
* LayoutTests/http/wpt/credential-management/setDigitalCredentialsEnable.https.html: Renamed from LayoutTests/http/wpt/credential-management/setDigitalIdentityEnable.https.html.
* LayoutTests/http/wpt/credential-management/setDigitalCreditialsEnable.https-expected.txt: Renamed from LayoutTests/http/wpt/credential-management/setDigitalIdentityEnable.https-expected.txt.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/credentialmanagement/CredentialRequestOptions.h:
* Source/WebCore/Modules/credentialmanagement/CredentialRequestOptions.idl:
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.idl:
* Source/WebCore/Modules/identity/Navigator+Identity.idl:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/275616@main">https://commits.webkit.org/275616@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2211983bb2e128e590e90be2cadb396002e38db1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42304 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21322 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44698 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44903 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38421 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/44611 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/24552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18664 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35049 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42878 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/18271 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36436 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15980 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15934 "layout-tests (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/37476 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/46361 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/35760 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/38514 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37806 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/41716 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/41932 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17120 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14104 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/40311 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18739 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/48941 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9468 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18801 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/9927 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18384 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/Modules/credentialmanagement/CredentialRequestOptions.idl:
```diff
@@ -30,4 +30,6 @@
     AbortSignal signal;
     PublicKeyCredentialRequestOptions publicKey;
+    // https://wicg.github.io/digital-identities/#extensions-to-credentialrequestoptions-dictionary
+    DigitalCredentialRequestOptions digital;
```
Is this dictionary already web-exposed? Because if so, I'm worried that now it will seem as if we support this new key, when we don't.

marcoscaceres replied:
Added Conditional.

annevk replied:
Why can't you use the runtime permission here? I don't think we need a compile flag.

marcoscaceres replied:
Sure, can switch it to `EnabledBySetting=DigitalCredentialsEnabled`... will make it easier to test too.

annevk commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -315,7 +315,10 @@
 		0C3F1F5B10C8871200D72CE1 /* WebGLUniformLocation.h in Headers */ = {isa = PBXBuildFile; fileRef = 0C3F1F5810C8871200D72CE1 /* WebGLUniformLocation.h */; };
 		0C45342810CDBBFA00869157 /* JSWebGLUniformLocation.h in Headers */ = {isa = PBXBuildFile; fileRef = 0C45342610CDBBFA00869157 /* JSWebGLUniformLocation.h */; };
 		0D431A7D2B5A397C00864655 /* DigitalCredential.h in Headers */ = {isa = PBXBuildFile; fileRef = 0D431A7C2B5A397C00864655 /* DigitalCredential.h */; };
+		0D431F002B9053DC001E222F /* DigitalCredentialRequestOptions.h in Headers */ = {isa = PBXBuildFile; fileRef = 0D494DEC2B5DF3DC00B61C51 /* DigitalCredentialRequestOptions.h */; settings = {ATTRIBUTES = (Private, ); }; };
```
I don't understand why this file is changed as you're not changing any files. Seems best to leave out of this PR?

marcoscaceres replied:
Those are needed to otherwise the DigitalCredentialRequestOptions.h doesn't end up in "private headers", and then it won't compile.

annevk replied:
Oh I see, it wasn't included before. Okay.

marcoscaceres committed:
CredentialRequestOptions gains a "digital" member
https://bugs.webkit.org/show_bug.cgi?id=270196
rdar://123744112

Reviewed by Anne van Kesteren.

Adds "digital" member to CredentialRequestOptions dictionary:
https://wicg.github.io/digital-identities/#extensions-to-credentialrequestoptions-dictionary

Given the spec changed names again, changed pref name to DigitalIdentityEnabled and updated a few files to reflect the name change.

* LayoutTests/http/wpt/credential-management/setDigitalCredentialsEnable.https-expected.txt: Copied from LayoutTests/http/wpt/credential-management/setDigitalIdentityEnable.https-expected.txt.
* LayoutTests/http/wpt/credential-management/setDigitalCredentialsEnable.https.html: Renamed from LayoutTests/http/wpt/credential-management/setDigitalIdentityEnable.https.html.
* LayoutTests/http/wpt/credential-management/setDigitalCreditialsEnable.https-expected.txt: Renamed from LayoutTests/http/wpt/credential-management/setDigitalIdentityEnable.https-expected.txt.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/credentialmanagement/CredentialRequestOptions.h:
* Source/WebCore/Modules/credentialmanagement/CredentialRequestOptions.idl:
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.idl:
* Source/WebCore/Modules/identity/Navigator+Identity.idl:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/275616@main
